### DISCUSSION TITLE
A time picker that isn't painful to use

### DISCUSSION THREAD

**Discussion by stamminator** [microsoft/microsoft-ui-xaml#10184] [Q&A]
The default TimePicker control is not suitable for desktop UI. It's probably fine for touch, but it punishes the majority of users (keyboard & mouse) by requiring them to scroll through dropdowns to select their time and not allowing them to type or paste. I find it unintuitive and frustrating to use. Compare this to the time input in Microsoft Teams or Outlook, which provides a dropdown of times, but still allows the user to type.

![Animation2](https://github.com/user-attachments/assets/4f7b52f9-3e03-46d9-9f16-99f5b39a499e)

Is there a pattern for using this control I'm not aware of that makes it appropriate for desktop use?

**Comment by kmgallahan** [comment]
> Is there a pattern for using this control I'm not aware of that makes it appropriate for desktop use?

Using the mouse wheel.

**Reply by stamminator** [reply]
The mouse wheel makes it _less_ painful, but not by much. It's still an interface optimized for touch at the expense of standard desktop usage. And if you're a keyboard & mouse user who likes to tab through form elements and minimize reaching for the mouse, the disparity in UI efficiency between these two styles of time pickers is even more stark.

When I said "using this control", I mean as a developer, not a user. I'm asking if anyone has already worked out a way of using the existing TimePicker control as part of a larger control which incorporates typing. Look no further than the Teams part of my screen recording for an example of this being done successfully.

**Comment by whiskhub** [comment]
I would advise to create a feature request in the repo, otherwise I doubt it will change.
